eng/pipelines/templates/jobs/ci.yml

```diff
@@ -225,36 +239,50 @@ jobs:
           arguments: 'restore'
           workingDirectory: '$(Agent.BuildDirectory)'
 
-      - task: PowerShell@2
-        displayName: "Verify generated code"
+      - pwsh: |
+          $failed = $false
+          $serviceDirectories = "$(ChangedServices)" -split ","
+          $ScanPaths = @()
+          foreach ($ServiceDirectory in $serviceDirectories) {
+            ./eng/scripts/CodeChecks.ps1 `
+              -ServiceDirectory $ServiceDirectory `
+              -SDKType ${{ parameters.SDKType }} `
+              -SpellCheckPublicApiSurface `
+
+            if ($LASTEXITCODE -ne 0) {
+              $failed = $true
+            }
+          }
+
+          if ($failed) {
+            exit 1
+          }
+        displayName: Verify Generated Code
         env:
           EnableSourceLink: false
-        inputs:
-          filePath: "eng/scripts/CodeChecks.ps1"
-          arguments: >
-            -ServiceDirectory ${{ parameters.ServiceDirectory }}
-            -SDKType ${{ parameters.SDKType }}
-            -SpellCheckPublicApiSurface
-          pwsh: true
-          failOnStderr: false
 
       - template: /eng/pipelines/templates/steps/dotnet-diagnostics.yml
         parameters:
           LogFilePath: $(Build.ArtifactStagingDirectory)/rebuild.binlog
 
-      - script: >-
-          dotnet build eng/service.proj -warnaserror
-          /t:rebuild
-          /p:DebugType=none
-          /p:SDKType=${{ parameters.SDKType }}
-          /p:ServiceDirectory=${{ parameters.ServiceDirectory }}
-          /p:IncludePerf=false
-          /p:IncludeStress=false
-          /p:PublicSign=false
-          /p:Configuration=$(BuildConfiguration)
-          /p:EnableSourceLink=false
-          /p:BuildSnippets=true
-          $(DiagnosticArguments)
+      - pwsh: |
+          $services = "$(ChangedServices)" -split ","
+
+          foreach ($service in $services) {
```

Comment:
Can we not use the project override list for this step?

Reply:
I think we can. There's another place in the build stage that does this as well.

Reply:
I'm going to take this as a follow-up improvement while I'm tackling the checks where a parameter needs to be set to enable. Specifically `aotcompat` and `snippet` assembly.